fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java

```diff
@@ -308,13 +312,15 @@ private OptExprBuilder castOutputColumnsTypeToTargetColumns(ColumnRefFactory col
                                                                 OptExprBuilder root) {
         List<Column> fullSchema = insertStatement.getTargetTable().getFullSchema();
         Map<ColumnRefOperator, ScalarOperator> columnRefMap = new HashMap<>();
-
+        ScalarOperatorRewriter rewriter = new ScalarOperatorRewriter();
+        List<ScalarOperatorRewriteRule> rewriteRules = Arrays.asList(new FoldConstantsRule());
```

Comment:
Why Optimizer.optimize cannot handle this? still add a rule here?

Reply:
> Why Optimizer.optimize cannot handle this? still add a rule here?

There seems to be no special rule processing for this scenario, it is a very local problem, so I just did rewrite here……do you have a better suggestion? @Seaven

Reply:
Optimizer can only handle query stmt. The target table of insert stmt is invisible for optimizer, so only handle it in here